rfc/0009-policy-report.md

```diff
@@ -0,0 +1,323 @@
+  Resources:
+    API Version: v1
+    Kind:        Pod
+    Name:        nginx-abuse
+    Namespace:   default
+    UID:         nginx-abuse-pod-uuid
+---
+  Source:  kubewarden
+  Policy:  ap-safe-labels
+  Result:  pass
+  Timestamp: <a timestamp>
+  Resources:
+    API Version: v1
+    Kind:        Pod
+    Name:        nginx-abuse
+    Namespace:   default
+    UID:         nginx-abuse-pod-uuid
+---
+  Source:  kubewarden
+  Policy:  cap-privileged-containers
+  Result:  pass
+  Timestamp: <a timestamp>
+  Resources:
+    API Version: v1
+    Kind:        Pod
+    Name:        docker-registry
+    Namespace:   default
+    UID:         docker-registry-pod-uuid
+---
+  Source:  kubewarden
+  Policy:  ap-safe-labels
+  Result:  fail
+  Message: the `hello-world` label is not allowed
+  Timestamp: <a timestamp>
+  Resources:
+    API Version: v1
+    Kind:        Pod
+    Name:        docker-registry
+    Namespace:   default
+    UID:         docker-registry-pod-uuid
+---
+  Source:  kubewarden
+  Policy:  ap-safe-labels
+  Result:  pass
+  Timestamp: <a timestamp>
+  Resources:
+    API Version: networking.k8s.io/v1
+    Kind:        Ingress
+    Name:        docker-registry
+    Namespace:   default
+    UID:         docker-registry-ingress-uuid
+---
+  Source:  kubewarden
+  Policy:  cap-ingress-lets-encrypt
+  Result:  fail
+  Message: wrong let's encrypt configuration is being used
+  Timestamp: <a timestamp>
+  Resources:
+    API Version: networking.k8s.io/v1
+    Kind:        Ingress
+    Name:        docker-registry
+    Namespace:   default
+    UID:         docker-registry-ingress-uuid
+```
+
+## Creating `ClusterPolicyReport` resource
+
+The `ClusterPolicyReport` object contains audit results for all the cluster wide
+Kubernetes resources.
+
+Since Kubewarden allows this kind of resources to be inspected only by
+`ClusterAdmissionPolicy` policies, the report will be built in the following way:
+
+* Find all the `ClusterAdmissionPolicy` policies that are interested about cluster wide
+  objects
+* Remove from this list all the `ClusterAdmissionPolicy` that are not interested
+  about `CREATE` events
+* Find all the cluster wide resources that are relevant to these policies
+* Evaluate the resources using the policies
+* Store the evaluation results inside of the `ClusterPolicyReport` object
+
+The CRD of `ClusterAdmissionPolicy` is basically equal to the one of `PolicyReport`.
+Because of that, we will use it in the same way as described in the previous
+section about `PolicyReport`.
+
+## Changes to `ClusterAdmissionPolicy` and `AdmissionPolicy`
+
+We will extend the `ClusterAdmissionPolicy` and `AdmissionPolicy` resources to have
+a new optional field called `background`.
+This field is going to hold a boolean value, which is going to be set to be `true` by default.
+
+Policies that have `background` set to false will be ignored during the background
+scans. Hence they will never show inside of the reports.
```

Comment:
I would expect to have `PolicyReports` information for non-background policy runs too, be it enforcing or monitoring, to be fair.

Reply:
That's a good point. I think it deserves to be mentioned in the RFC. Better still if we document the requests that will not be faked/audited, to provide to the users.

Reply:
[raised in the other RFC](https://github.com/kubewarden/rfc/pull/10#discussion_r981237592), resolving this.